api_spec.yaml

```diff
@@ -1266,17 +1472,285 @@ paths:
+            responses:
+                200:
+                    description: Success
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+                                properties:
+                                    status:
+                                        type: string
+                                        default: "OK"
+                                    wasAssociated:
+                                        type: boolean
+                                        default: true
+
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+    /dashboard/api/tenants/third-party:
+        put:
+            tags:
+                - Multitenancy
+            summary: Create or update third party config for a tenant
+            parameters:
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                tenantId:
+                                    type: string
+                                providerConfig:
+                                    $ref: "#/components/schemas/thirdPartyProvider"
+            responses:
+                200:
+                    description: Success
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+                                properties:
+                                    status:
+                                        type: string
+                                        default: "OK"
+                                    createdNew:
+                                        type: boolean
+                                        default: true
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+        delete:
+            tags:
+                - Multitenancy
+            summary: Delete thirdParty config for a tenant
+            parameters:
+                - name: tenantId
+                  in: query
+                  required: true
+                  schema:
```

Comment:
What if tenant was not found? check similar issue in other thirdparty config related APIs

Reply:
We are not handling it separately in the backend SDK and let the core return the error accordingly, should I check if tenant exists or not and then return the status accordingly?

Reply:
Check what's the backend sdk behaviour, does it return null or throws an error?